src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/IngestionFileHandlerService.java

```diff
@@ -0,0 +1,71 @@
+package it.gov.pagopa.payhub.activities.service.ingestionflow;
+
+import it.gov.pagopa.payhub.activities.util.AESUtils;
+import it.gov.pagopa.payhub.activities.util.FileUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Service class responsible for handling encrypted ingestion files,
+ * including decryption, validation, and extraction of ZIP files.
+ */
+@Slf4j
+@Service
+public class IngestionFileHandlerService {
+	private static final String TEMPORARY_PATH = "TEMP";
+
+	/**
+	 * The password used for decrypting encrypted files.
+	 */
+	private final String dataCipherPsw;
+
+		public IngestionFileHandlerService(@Value("${data-cipher.encrypt-psw}") String dataCipherPsw) {
+		this.dataCipherPsw = dataCipherPsw;
+	}
+
+	/**
+	 * Handles the setup process for an ingestion file by performing the following steps:
+	 * <ul>
+	 *     <li>Decrypts the file using AES encryption.</li>
+	 *     <li>Validates if the file is a valid ZIP archive.</li>
+	 *     <li>Extracts the contents of the ZIP file to a temporary directory.</li>
+	 * </ul>
+	 *
+	 * @param relativePath the relative path to the directory containing the file.
+	 * @param filename the name of the file to process.
+	 * @return the path to the extracted XML file.
+	 * @throws IOException if any file operation fails during the setup process.
+	 */
+	public Path setUpProcess(String relativePath, String filename) throws IOException {
+		Path relativePathDir = Paths.get(relativePath);
+		Path encryptedFilePath = relativePathDir.resolve(filename);
+		FileUtils.validateFile(encryptedFilePath);
+
+		Path temporaryPath = relativePathDir.resolve(TEMPORARY_PATH);
```

Comment:
if TEMPORARY_PATH will be `Paths.of("/tmp")`, this could be
TEMPORARY_PATH.resolve(relativePathDir.subpath(0, relativePathDir.getNameCount()))

Reply:
According to the legacy system, the temporary folder was always at the end and not at the root

Reply:
The change proposed is not a business logic change, but a technical requirement.
The current directory is inside a share, let's avoid to overuse it in order to:
* Reduce the network overhead
* Avoid to expose plain data
* Avoid to dirty the folder with temporary files of interest of just the running pod